src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp

```diff
@@ -50,26 +52,31 @@ auto BasisImageConverter::doFeatures() const -> Features { return Feature::Conve
 Containers::Array<char> BasisImageConverter::doExportToData(const ImageView2D& image) {
     /* Check input */
     if(image.format() != PixelFormat::RGB8Unorm
-        && image.format() != PixelFormat::RGBA8Unorm) {
-        Error() << "Trade::BasisImageConverter::exportToData(): unsupported pixel format" << image.format();
+        && image.format() != PixelFormat::RGBA8Unorm
+        && image.format() != PixelFormat::RG8Unorm
+        && image.format() != PixelFormat::R8Unorm) {
+        Error() << "Trade::BasisImageConverter::exportToData(): unsupported format" << image.format();
         return {};
     }
 
     /* Magnum specific configuration */
-    const Int threadCount = configuration().value<Int>("num_threads");
-    if(threadCount < 1) {
-        Error() << "Trade::BasisImageConverter::exportToData(): invalid configuration: num_threads is expected to be at least 1, but was:" << threadCount;
+    Int threadCount = configuration().value<Int>("threads");
+    if(threadCount == 0) {
+        threadCount = std::thread::hardware_concurrency();
+    } else if(threadCount < 1) {
+        Error() << "Trade::BasisImageConverter::exportToData(): invalid configuration: threads is expected to be at least 0, but was"
+                << threadCount;
         return {};
```

Comment:
You could fetch the `threads` as `UnsignedInt` and don't do any range checking at all, then people who accidentally say `-1` would spin up 4 billion threads and fry their machines, which is not our problem. Same would happen if people enter `threads=400`.

Reply:
It seems it's not great user-experience to let them figure out that "-1" was casted to unsigned int, probably because they accidentally put a `-` there?

Reply:
i don't think it's our responsibility to check for that, same happens with `new int[-1];`, `malloc(-1)`, `vec.resize(-1)`...

Reply:
Hehe :D Fancy, aparently in my over then years with C++ I never actually came across this mistake 😂